Introducing 'contributor' type (release contributors info)

mickeyn opened the pull request:
New type and supporting code & test to fill/fetch the
release contributors info to/from the index.

* MetaCPAN::Script::Mapping::CPAN::Contributor
Added for type mapping for Elasticseach

* Script::Contributor
Added for data filling of release contributors.
Script can run in different modes:
--all          # run for all releases in the 'release' type
--distribution # run for all releases matching a given distribution
--release      # run for a single release (format: PAUSEID/DISTRIBUTION-VERSION)

* Script::Release
Updated to store contributor info for processed releases

* Script::Role::Contributor
Added to support common functionality shared by the scripts
for handling contributors info fetching/updating.

* Document::Contributor
Model representation of the new type.

* Server::Controller::Contributor
Controller endpoints for the new type.

mickeyn commented:
with a small change, I can make this new type live in its own isolated index... would you prefer this? (I think it's better as long as we don't need to share alias, which so far I don't think we do)

oalders commented on lib/MetaCPAN/Script/Contributor.pm:
```diff
@@ -0,0 +1,118 @@
+use warnings;
+
+BEGIN {
+    $ENV{PERL_JSON_BACKEND} = 'JSON::XS';
```
Can you remind me why we're doing this?

mickeyn replied:
oops, here it was copy-pasted from Script::Release... there it was added in 2f6d07b5e2... not sure how relevant it is.
i'll remove it from this script.

oalders commented on lib/MetaCPAN/Script/Role/Contributor.pm:
```diff
@@ -0,0 +1,68 @@
+use Moose::Role;
+
+use MetaCPAN::Util qw< digest >;
+use Ref::Util qw< is_arrayref >;
```
Nitpick, but could we just stick with round brackets?

mickeyn replied:
sure

oalders commented on t/server/controller/contributor.t:
```diff
@@ -0,0 +1,62 @@
+use Test::More;
+
+my $server = MetaCPAN::TestServer->new;
+$server->index_packages;
```
`index_packages` is already called in `t/00_setup.t:`  Do we need to do it again here?

mickeyn replied:
copy-pasted from the package test... not sure why it's there either - removing both.

mickeyn replied:
oh, origin is t/server/controller/permission.t:
```
my $server = MetaCPAN::TestServer->new;
$server->index_permissions;
```

oalders commented:
> with a small change, I can make this new type live in its own isolated index... would you prefer this? (I think it's better as long as we don't need to share alias, which so far I don't think we do)

Works for me.  At the very least if there are bugs, this would be a good place to have them.

mickeyn commented:
@oalders I fixed as requested and updated to support separate index (makes more sense to me)
I created the new index & mapping - so it should be safe to merge.
